build/config/compiler/BUILD.gn

```diff
@@ -671,3 +671,9 @@ config("specs_default") {
     ldflags = cflags
   }
 }
+
+config("link_as_needed") {
+  if (current_os == "linux" && is_clang) {
```

Comment:
Why restricted only to clang?

Reply:
I have instead blacklisted mac and ios (as I assume it also uses clang to build)